src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -3341,50 +3348,24 @@ public DnsNameProperty(X509Certificate2 cert)
             if (cert.Subject.StartsWith(distinguishedNamePrefix, System.StringComparison.OrdinalIgnoreCase) &&
                 !cert.Subject.Contains(','))
             {
-                name = cert.Subject.Substring(distinguishedNamePrefix.Length);
-                try
-                {
-                    unicodeName = idnMapping.GetUnicode(name);
-                }
-                catch (System.ArgumentException)
-                {
-                    // The name is not valid punyCode, assume it's valid ascii.
-                    unicodeName = name;
-                }
-
-                dnsName = new DnsNameRepresentation(name, unicodeName);
+                string parsedSubjectDistinguishedDnsName = cert.Subject.Substring(distinguishedNamePrefix.Length);
+                DnsNameRepresentation dnsName = GetDnsNameRepresentation(parsedSubjectDistinguishedDnsName);
```

Comment:
@iSazonov I wonder if this can also be replaced with https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x500distinguishedname.enumeraterelativedistinguishednames?view=net-9.0.

I don't think even distinguished name needs string parsing anymore.

Reply:
Actually will park this, just realised this will include all distinguished names outside of `CN=`. dotnet runtime probably needs to expose something for just common name without too much extra work. 

https://github.com/dotnet/runtime/issues/33914

Reply:
I think it would end up being something like:

```csharp
private const string CommonNameOid = "2.5.4.3";

// Extract DNS name from Subject distinguished name
foreach (X500RelativeDistinguishedName distinguishedName in cert.SubjectName.EnumerateRelativeDistinguishedNames())
{
   if (!distinguishedName.HasMultipleElements &&
        distinguishedName.GetSingleElementType().Value.Equals(CommonNameOid, StringComparison.OrdinalIgnoreCase))
    {
        DnsNameRepresentation dnsName = GetDnsNameRepresentation(distinguishedName.GetSingleElementValue());
        _dnsList.Add(dnsName);
    }
}
```

Reply:
The change is not related the PR. Please revert.
If you think original code has an issue, please open new issue to discuss.

Reply:
Yeah so this was just referring to how distinguished names are handled. I have reverted that change and will open a new issue to discuss. This is resolved for this PR.

Reply:
I don't see new commit.

Reply:
I force pushed and removed the changes.